src/app/triggerFromForm.ts

```diff
@@ -0,0 +1,115 @@
+
+    constructor(values: AppCallValues, context: AppContext) {
+        this.values = values;
+        this.context = context;
+        this.trigger = {};
+    }
+
+    getTrigger(): any {
+        // If not a new trigger, add the trigger ID to the payload
+        // This is a signal to update the trigger, not create a new one
+        if (!this.isNewTrigger()) {
+            const subID = this.values[SubscriptionFields.SubSelect_Name].value;
+            this.addField('id', subID);
+        }
+        this.addTitle();
+        this.addDescription();
+        this.addActions();
+        this.addConditions();
+
+        return {trigger: this.trigger};
+    }
+
+    addActions(): void {
+        const actions = [
+            {
+                field: TriggerFields.ActionField,
+                value: [
+                    TriggerFields.TargetID,
+                    this.getJSONDataFields(),
+                ],
+            },
+        ];
+        this.addField('actions', actions);
+    }
+
+    // getJSONDataFields constructs the object text string for a trigger
+    getJSONDataFields(): string {
+        const channelID = this.values[SubscriptionFields.ChannelPickerSelect_Name].value;
+        const pairs = TriggerFields.ActionValuePairs;
+        pairs[TriggerFields.ChannelIDKey] = channelID;
+        return JSON.stringify(pairs);
+    }
+
+    getChannelIDobject(): {} {
+        const fieldName = SubscriptionFields.ChannelPickerSelect_Name;
+        const channelID = this.values[fieldName].value;
+        return {fieldName: channelID};
+    }
+
+    addTitle(): void {
+        let title = SubscriptionFields.PrefixTriggersTitle;
+        title += this.context.channel_id;
+        title += ' ' + this.values[SubscriptionFields.SubText_Name];
+        this.addField('title', title);
+    }
+
+    addDescription(): void {
+        const description = String(this.values[SubscriptionFields.SubText_Name]);
+        this.addField('description', description);
+    }
+
+    addField(key: string, value: any): void {
+        this.trigger[key] = value;
+    }
+
+    isNewTrigger(): boolean {
+        const subID = this.values[SubscriptionFields.SubSelect_Name].value;
+        return subID === SubscriptionFields.NewSub_OptionValue;
+    }
+
+    addConditions(): void {
+        const conditions = {
+            any: [],
+        };
+        for (const checkbox of SubscriptionFields.ConditionsCheckBoxFields) {
+            if (this.values[checkbox]) {
+                const entry = {
+                    field: checkbox,
+                    operator: 'changed',
+                };
+                conditions.any.push(entry);
+            }
+        }
+
+        if (this.values && this.values[SubscriptionFields.SubmitButtonsName] === SubscriptionFields.DeleteButtonLabel) {
+            this.addField('conditions', conditions);
+        }
+
+        // do not let subscriptions without a condition be created...
+        if (conditions.any.length === 0) {
```

Comment:
should this be moved immediately after the `for`?

Reply:
I actually removed the code between this if block and the for block.  I don't recall the need for that code and think it's invalid